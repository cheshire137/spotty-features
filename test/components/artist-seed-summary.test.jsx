import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import ArtistSearchResponse from '../fixtures/spotify/artist-search'

import ArtistSeedSummary from '../../src/components/artist-seed-summary.jsx'

const artist = ArtistSearchResponse.artists.items[0]
const image = artist.images[3].url

function props() {
  return { image, name: artist.name }
}

describe('ArtistSeedSummary', () => {
  let component = null

  beforeEach(() => {
    component = <ArtistSeedSummary {...props()} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('displays artist name', () => {
    const title = shallow(component).find('.track-name')
    expect(title.text()).toBe('Tom Petty')
  })

  test('displays artist art', () => {
    const images = shallow(component).find('.track-image')
    expect(images.length).toBe(1)
    expect(images.at(0).props().src).toBe(image)
  })
})
