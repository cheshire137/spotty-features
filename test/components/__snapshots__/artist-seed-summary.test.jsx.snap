exports[`ArtistSeedSummary matches snapshot 1`] = `
<div
  className="columns seed-summary">
  <div
    className="column track-image-column">
    <img
      className="track-image"
      src="https://i.scdn.co/image/35033e0288ffdc1b0a0958cc917a93d5a28b8b8b" />
  </div>
  <div
    className="column">
    <span
      className="track-name">
      Tom Petty
    </span>
  </div>
</div>
`;
